app/labor/markdown_parser.rb

```diff
@@ -2,6 +2,12 @@ class MarkdownParser
   include ApplicationHelper
   include CloudinaryHelper
 
+  RENDER_OPTIONS = {
+    hard_wrap: true,
+    filter_html: false,
+    with_toc_data: true
+  }.freeze
```

Comment:
This is probably what's causing the error. My hunch is that `RENDER_OPTIONS` is modified in the process.

Reply:
Makes sense.

```
Validation failed: can't modify frozen Hash, Title can't be blank
```

Reply:
Indeed, freezing the object kills it. I'm in the process of updating the specs and adding a few extra test cases

Reply:
FYI I made a shared object just to DRY up the code. If we are going to unfreeze this object, you may want to go back to feeding in the options manually at each call site, so that you con't have a state error due to sharing a mutable object between multiple function calls. It depends on the precise way the app / libraries involved work.

This is a classic case of implicit state / mutation making it harder to compose things well… 😉

Reply:
huh.. looks like rubocop is automatically freezing the hash. weird ?

Reply:
it's because of the pre-commit hook. you can avoid with by appending `--no-verify` to back of your `git commit`